src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -1078,13 +1168,23 @@ public static unsafe Vector128<uint> Create(Vector64<uint> lower, Vector64<uint>
         [CLSCompliant(false)]
         public static unsafe Vector128<ulong> Create(Vector64<ulong> lower, Vector64<ulong> upper)
         {
-            Vector128<ulong> result128 = Vector128<ulong>.Zero;
+            if (AdvSimd.IsSupported)
+            {
+                return lower.ToVector128().WithUpper(upper);
+            }
+
+            return SoftwareFallback(lower, upper);
+
+            static Vector128<ulong> SoftwareFallback(Vector64<ulong> lower, Vector64<ulong> upper)
```

Comment:
Can you explain this pattern of segregating the code into a local method? If it's to reduce the size of the outer method, it guaranteed that future codegen will never inline it?

Reply:
Or is it because you can rely on codegen for the outer method first resolving `AdvSimd.IsSupported` to a constant true such that it knows the call to the local method will be dead and it will never be considered for inlining.

If that's the case, would ideally this pattern not be needed -- it could just stop generating any code after it encountered the unconditional return above.

Reply:
This pattern is done because the containing method is typically marked with `AggressiveInlining`.

The codegen for the accelerated case is generally 1-2 instructions and so it is something we want to always inline and the codegen for all the paths + the fallback typically makes the method to big for that to happen naturally. The codegen for the fallback case, however, is often of unknown size or is used in scenarios where inlining might not be beneficial and so we want the JIT to make the inlining decision on its own.

So, this pattern ensures that the accelerated case is always inlined and that the fallback case is ultimately inlined at the JITs disgression. Due to the containing method being `AggressiveInlining`, the "worst" case is a direct call to `SoftwareFallback`.